### PR TITLE
Feature mosip 12742

### PR DESCRIPTION
**Pull request by urviljoshi**

### CODE REVIEW

File: kernel/kernel-auth-service/src/main/java/io/mosip/kernel/auth/defaultimpl/repository/DataStore.java [hunk 1]
```diff
@@ -64,5 +65,7 @@
 	public ValidationResponseDto validateUserName(String userId);
 
 	public UserDetailsResponseDto getUserDetailBasedOnUid(List<String> userIds);
+	
+	public VidDto getVidFromUserId(String userId,String realmID);
```
Comment (loganathan-sekaran) [inline kernel/kernel-auth-service/src/main/java/io/mosip/kernel/auth/defaultimpl/repository/DataStore.java:69]: @urviljoshi Can we rename the realmID as appID

File: kernel/kernel-auth-service/src/main/java/io/mosip/kernel/auth/defaultimpl/repository/impl/KeycloakImpl.java [hunk 2]
```diff
@@ -690,4 +691,42 @@ private String getRoleId(String roleName,String realmId) throws IOException {
...
+			for (JsonNode jsonNode : node) {
+				if (jsonNode.get(AuthConstant.USER_NAME).textValue().equals(userId)) {
+					JsonNode attriNode = jsonNode.get("attributes");
+					String vid = attriNode.get(AuthConstant.VID).get(0).textValue();
+					vIDDto.setVid(vid);
+ 					break;
+				}
+			}
+			if (vIDDto.getVid() == null) {
+				throw new AuthManagerException(AuthErrorCode.USER_NOT_FOUND.getErrorCode(),
```
Comment (loganathan-sekaran) [inline kernel/kernel-auth-service/src/main/java/io/mosip/kernel/auth/defaultimpl/repository/impl/KeycloakImpl.java]: @urviljoshi If the user is found but the RID attribute is not found, can we return a different message like 'RID attribute is not available for the user' ? Returning the 'user not found' in such case will be misleading.

Comment (urviljoshi) [reply kernel/kernel-auth-service/src/main/java/io/mosip/kernel/auth/defaultimpl/repository/impl/KeycloakImpl.java]: sure

File: kernel/kernel-auth-service/src/main/java/io/mosip/kernel/auth/defaultimpl/repository/DataStore.java (hunk 1 above)
Comment (urviljoshi) [reply kernel/kernel-auth-service/src/main/java/io/mosip/kernel/auth/defaultimpl/repository/DataStore.java:69]: @LoganathanSekar7627  in service layer we get realmid from app id  it is named as appid in service layer when we send it to repo we have realm id .

**Review by loganathan-sekaran — APPROVED** [review]
Approved

**Pull request merged**